huaweicloud/services/dws/resource_huaweicloud_dws_disaster_recovery.go

```diff
@@ -0,0 +1,543 @@
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"standby_cluster_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"dr_sync_period": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"action": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+			"status": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"started_at": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"created_at": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"primary_cluster": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem:     clusterSchema(),
+			},
+			"standby_cluster": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem:     clusterSchema(),
+			},
+		},
+	}
+}
+
+func clusterSchema() *schema.Resource {
+	nodeResource := schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"region": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"cluster_az": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"role": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"status": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"progress": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"last_success_at": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"obs_bucket_name": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+
+	return &nodeResource
+}
+
+func resourceDwsDisasterRecoveryCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	var (
+		cfg     = meta.(*config.Config)
+		region  = cfg.GetRegion(d)
+		httpUrl = "v2/{project_id}/disaster-recoveries"
+		product = "dws"
+	)
+
+	client, err := cfg.NewServiceClient(product, region)
+	if err != nil {
+		return diag.Errorf("error creating DWS client: %s", err)
+	}
+	createPath := client.Endpoint + httpUrl
+	createPath = strings.ReplaceAll(createPath, "{project_id}", client.ProjectID)
+
+	createOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+		JSONBody:         utils.RemoveNil(buildCreateDisasterRecoveryBodyParams(d)),
+		MoreHeaders:      requestOpts.MoreHeaders,
+	}
+	resp, err := client.Request("POST", createPath, &createOpt)
+	if err != nil {
+		return diag.Errorf("error creating DWS disaster recovery: %s", err)
+	}
+	respBody, err := utils.FlattenResponse(resp)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	id, err := jmespath.Search("disaster_recovery.id", respBody)
+	if err != nil {
+		return diag.Errorf("error creating DWS disaster recovery: ID is not found in API response")
+	}
+	d.SetId(id.(string))
+
+	err = waitingForActionDisasterRecvery(ctx, d, client, "unstart")
```

Comment:
Add a comment that when the resource is successfully created, the status is "unstart".